apps/docs/hooks/use-table-content-observer.tsx

```
@@ -24,12 +24,9 @@ export const useTableContentObserver = (setActiveId: (id: string) => void, pathn
   useEffect(() => {
     const callback = (headings: HeadingElement[]) => {
       // Create a map of heading elements, where the key is the heading's ID and the value is the heading element
-      headingElementsRef.current = headings.reduce(
-        (map, headingElement) => {
-          return { ...map, [headingElement.target.id]: headingElement };
-        },
-        {}
-      );
+      headingElementsRef.current = headings.reduce((map, headingElement) => {
+        return { ...map, [headingElement.target.id]: headingElement };
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)